src/action_worker.ts

```diff
@@ -20,10 +20,14 @@ let isRunning = false;
 await startTask();
 
 self.onmessage = (evt) => {
-  const job = evt.data as Job;
-  queue.addJob(job);
-  if (!isRunning) startTask();
-  else console.log("Already running");
+  const job = evt.data as Job | "FULLUPDATE";
+  if (job === "FULLUPDATE") {
+    gatherJobsForFullUpdate();
+  } else {
+    queue.addJob(job);
+    if (!isRunning) startTask();
```

Comment:
shouldn't this be called outside the else? (i.e in any case)

Reply:
Okay, I see what you're doing. I'm a bit skeptical about the re-use of `isRunning`. Meditating on it.

Reply:
`startTask()` is called  by `gatherJobsForFullUpdate()`, this way the message handler doesn't need to await `gatherJobsForFullUpdate()`.

`isRunning` blocks jobs from running while gathering, to ensure that the full update runs on a consistent initial state.